manifests/config.pp

```diff
@@ -6,14 +6,24 @@
 
   assert_private()
 
-  file { $::telegraf::config_file:
-    ensure  => file,
-    content => template('telegraf/telegraf.conf.erb'),
-    mode    => '0640',
-    owner   => 'telegraf',
-    group   => 'telegraf',
-    notify  => Class['::telegraf::service'],
-    require => Class['::telegraf::install'],
+  file {
```

Comment:
Yes, probably a good idea to maintain compatibility with older versions of Puppet for now.